pkg/opscomments/comments.go

```diff
@@ -0,0 +1,161 @@
+package opscomments
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+)
+
+var (
+	testAllRegex      = regexp.MustCompile(`(?m)^/test\s*$`)
+	retestAllRegex    = regexp.MustCompile(`(?m)^/retest\s*$`)
+	testSingleRegex   = regexp.MustCompile(`(?m)^/test[ \t]+\S+`)
+	retestSingleRegex = regexp.MustCompile(`(?m)^/retest[ \t]+\S+`)
+	oktotestRegex     = regexp.MustCompile(`(?m)^/ok-to-test\s*$`)
+	cancelAllRegex    = regexp.MustCompile(`(?m)^(/cancel)\s*$`)
+	cancelSingleRegex = regexp.MustCompile(`(?m)^(/cancel)[ \t]+\S+`)
+)
+
+type EventType string
+
+func (e EventType) String() string {
+	return string(e)
+}
+
+var (
+	NoOpsCommentEventType        = EventType("no-ops-comment")
+	TestCommentEventType         = EventType("test-comment")
+	RetestSingleCommentEventType = EventType("retest-comment")
+	RetestAllCommentEventType    = EventType("retest-all-comment")
+	OnCommentEventType           = EventType("on-comment")
```

Comment:
Can you add a comment for `on-comment` purpose as well as some unit test how user can pass `on-comment` ?

Reply:
yeah it's following PR, it was too annoying to break down all refactoring and implementations.. but that's fine since it's coming down..